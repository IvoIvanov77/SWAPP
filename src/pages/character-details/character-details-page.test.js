import React from 'react';
import { shallow } from 'enzyme';
import ChatacterDetaislPage from './character-details-page.component';

describe('ChatacterDetaislPage component', () => {
  let wrapper;
  let mockPerson;
  const mockedStarships = [{ id: 1 }, { id: 2 }];
  const mockName = 'name';

  beforeEach(() => {
    mockPerson = {
      name: mockName,
      starships: mockedStarships,
    };

    wrapper = shallow(<ChatacterDetaislPage person={mockPerson} />);
  });

  it('should render ChatacterDetaislPage component', () => {
    console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });

  it('should render character name', () => {
    const characterName = wrapper.find('CharacterName');
    expect(characterName.text()).toBe(mockName);
  });

  it('should render character info', () => {
    const characterInfo = wrapper.find('CharacterInfo');
    expect(characterInfo.prop('person')).toBe(mockPerson);
  });

  it('should render starships list', () => {
    const starshipsList = wrapper.find('StarshipList');
    expect(starshipsList.prop('starships')).toBe(mockedStarships);
  });
});
