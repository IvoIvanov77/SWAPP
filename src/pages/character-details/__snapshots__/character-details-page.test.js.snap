// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatacterDetaislPage component should render ChatacterDetaislPage component 1`] = `ShallowWrapper {}`;
